python/feldera/sql_context.py

```diff
@@ -0,0 +1,278 @@
+        return self
+
+    def register_table(self, table_name: str, schema: Optional[SQLSchema] = None, ddl: str = None):
+        """
+        Registers a table with the SQLContext. The table can be registered with a schema or with the SQL DDL.
+        One of the two must be provided, but not both.
+        Auto inserts the trailing semicolon if not present.
+        In the future, schema will be inferred from the data provided from applicable sources.
+
+        :param table_name: The name of the table.
+        :param schema: The schema of the table.
+        :param ddl: The SQL DDL of the table.
+        """
+
+        if not schema and not ddl:
+            raise ValueError("Schema inference isn't supported yet, either provide a schema or the SQL DDL")
+
+        if schema and ddl:
+            raise ValueError("Provide either a schema or the SQL DDL, not both")
+
+        if ddl:
+            self.register_table_from_sql(ddl)
+            return
+
+        if schema:
+            self.tables[table_name] = SQLTable(table_name, schema=schema)
+        else:
+            self.todo_tables[table_name] = None
+
+    def register_table_from_sql(self, ddl: str):
+        """
+        Registers a table with the provided SQL DDL.
+        Auto inserts the trailing semicolon if not present.
+
+        :param ddl: The SQL DDL of the table.
+        """
+
+        if ddl[-1] != ';':
+            ddl += ';'
+
+        name = _table_name_from_sql(ddl)
+
+        self.tables[name] = SQLTable(name, ddl)
+
+    def input_from_pandas(self, table_name: str, df: pandas.DataFrame):
+        """
+        Adds a pandas DataFrame to the input buffer of the SQLContext, to be pushed to the pipeline.
+
+        :param table_name: The name of the table.
+        :param df: The pandas DataFrame to be pushed to the pipeline.
+        """
+
+        tbl = self.tables.get(table_name)
+
+        if tbl:
+            # tbl.validate_schema(df)   TODO: something like this would be nice
+            self.http_input_buffer.append({tbl.name: df.to_dict('records')})
+            return
+
+        tbl = self.todo_tables.get(table_name)
+
+        if not tbl:
+            raise ValueError(f"Table {table_name} not registered")
+
+        # tbl.infer_schema(df)  TODO: support schema inference
+
+        self.tables[table_name] = tbl
+        self.todo_tables.pop(table_name)
+
+        self.http_input_buffer.append({tbl.name: df.to_dict('records')})
+
+    def register_view(self, name: str, query: str):
+        """
+        Registers a Feldera View based on the provided query.
+        Auto inserts the trailing semicolon if not present.
+
+        :param name: The name of the view.
+        :param query: The query to be used to create the view.
+        """
+
+        if query[-1] != ';':
+            query += ';'
+
+        self.views[name] = f"CREATE VIEW {name} AS {query}"
+
+    def listen(self, view_name: str) -> OutputHandler:
```

Comment:
When running in streaming mode, we will need a way to start/stop listening to a view at runtime.